interface/forms/vitals/C_FormVitals.class.php

```diff
@@ -150,20 +181,83 @@ public function default_action_process()
-            $_POST['process'] = "";
+        // have to set these global settings in order for us to save.
+        $obj->set_encounter($GLOBALS['encounter']);
+        $obj->set_pid($GLOBALS['pid']);
+        $obj->set_authorized($_SESSION['userauthorized']);
+
+        // handle all of the vital details that we need here.
+        $detailsToUpdate = array();
+        if (isset($_POST['interpretation'])) {
+            $interpretationList = $this->get_interpretation_list_as_hash();
+            // grab our default list options
+            foreach ($_POST['interpretation'] as $column => $value) {
+                $details = $this->vitals->get_details_for_column($column) ?? new FormVitalDetails();
+
+                // if we have nothing set and nothing saved we are going to leave these values as empty
+                if (empty($value) && empty($details->get_id())) {
+                    continue;
+                }
+
+                if (empty($value)) {
+                    $details->clear_interpretation();
+                } else if (isset($interpretationList[$value])) {
+                    $interpretation = $interpretationList[$value];
+
+                    // we save off both the code and the text value here which duplicates the data.  However, since
+                    // users can edit the code / text values of these values through list_options we have to have
+                    // a historical record so we save these off
+                    // TODO: change this if list_options ever saves historical records and keeps the id uniquely the same
+                    // then we can remove code/text
+                    $details->set_interpretation_list_id(FormVitals::LIST_OPTION_VITALS_INTERPRETATION);
+                    $details->set_interpretation_option_id($value);
+                    $details->set_interpretation_codes($value); // for now the option_id is the code
+                    $details->set_interpretation_title($interpretation['title']);
+                } else {
+                    (new SystemLogger())->error(
+                        "Passed in interpretation does not exist in list options, clearing interpretation id",
+                        ['form_id' => $this->vitals->get_id(), 'column' => $column, 'interpretation' => $value]
+                    );
+                    $details->clear_interpretation();
+                }
+
+                $details->set_vitals_column($column);
+                $detailsToUpdate[$column] = $details;
+            }
         }
 
-        return;
+        foreach ($detailsToUpdate as $column => $details) {
+            $this->vitals->set_details_for_column($column, $details);
+        }
+    }
+
+    private function populate_session_user_information(FormVitals $vitals)
+    {
+        $vitals->set_groupname($_SESSION['authProvider']);
+        $vitals->set_user($_SESSION[$_SESSION['authUser']]);
```

Comment:
this seems odd: `$_SESSION[$_SESSION['authUser']]`

Reply:
Interesting.. almost looks like it should never work... Good catch.

Reply:
I got this working.  Interestingly I noticed that group and user were not saving even in master for the vitals_form (but were saving in the forms table).  Fixed it in both places.